utils/windowfunction/wf_sum_avg.cpp

```diff
@@ -188,31 +167,19 @@ boost::shared_ptr<WindowFunctionType> WF_sum_avg<T_IN, T_OUT>::makeFunction(int
         }
 
         case CalpontSystemCatalog::DECIMAL:
+        case CalpontSystemCatalog::UDECIMAL:
         {
-            if (wc->functionParms()[0]->resultType().colWidth < 16)
+            if (wc->functionParms()[0]->resultType().colWidth < datatypes::MAXDECIMALWIDTH)
             {
                 func.reset(new WF_sum_avg<int64_t, int128_t>(id, name));
             }
-            else
+            else if (wc->functionParms()[0]->resultType().colWidth == datatypes::MAXDECIMALWIDTH)
             {
                 func.reset(new WF_sum_avg<int128_t, int128_t>(id, name));
             }
             break;
         }
         
-        case CalpontSystemCatalog::UDECIMAL:
-        {
-            if (wc->functionParms()[0]->resultType().colWidth < 16)
-            {
-                func.reset(new WF_sum_avg<uint64_t, uint128_t>(id, name));
-            }
-            else
-            {
-                func.reset(new WF_sum_avg<uint128_t, uint128_t>(id, name));
-            }
-            break;
-        }
-
```

Comment:
Please see my previous comment on verifying the correctness of removing this case block for UDECIMAL.

@dhall-MariaDB FYI.